### PR TITLE
Rolling interval

### LINKED ISSUE #32: Basic rolling window on timeseries
Rolling window defined by an time interval, allowing to calculate the following functions on columns:
* mean 
* median
* std

### PR DESCRIPTION
**Pull request by mwu-tow**
Added rolling interval functionality that allows calculating a number of functions for groups of samples defined by fitting a time interval window.
Fixes #32 #36 #44
Partially implements #35

### CODE REVIEW

File: native_libs/src/Core/ArrowUtilities.cpp [hunk 1]
```diff
@@ -27,6 +27,18 @@ std::vector<std::shared_ptr<arrow::Column>> getColumns(const arrow::Table &table
...
 
+std::shared_ptr<arrow::Column> getColumn(const arrow::Table &table, std::string_view name)
+{
+    auto columns = getColumns(table);
+    for(auto &col : columns)
+        if(col->name() == name)
+            return col;
+
+    auto names = transformToVector(columns, [](auto &col) { return col->name(); });
+    THROW("Failed to found column by name `{}`. Available columns: `{}`",
```
Comment (mikusp) [inline native_libs/src/Core/ArrowUtilities.cpp]: Failed to find column by name

File: src/Internal/CWrappers.luna [hunk 2]
```diff
@@ -373,7 +373,20 @@ class TableWrapper:
             Array CInt32 . with aggregationFunctionCounts aggregateFunctionCountsC:
                 bracket (aggregationFunctionIds.each (Array CInt8 . fromList _)) (_.each .free) listOfAggregatedFunctionCArrays:
                     Array (Pointer CInt8) . with (listOfAggregatedFunctionCArrays.each .ptr) arrayOfArraysWithIds:
-                        callHandlingError "tableAggregateBy" (Pointer None) [self.ptr.toCArg, keyColumn.ptr.toCArg, CInt32.fromInt aggregation.length . toCArg, aggregatedColumnsC.toCArg, aggregateFunctionCountsC.toCArg, arrayOfArraysWithIds.toCArg]
+                        callHandlingError "tableAggregateBy" (Pointer None) [keyColumn.ptr.toCArg, CInt32.fromInt aggregation.length . toCArg, aggregatedColumnsC.toCArg, aggregateFunctionCountsC.toCArg, arrayOfArraysWithIds.toCArg]
+        wrapReleasableResouce TableWrapper ptr
+
+    # aggregateBy :: ColumnWrapper -> [(ColumnWrapper, [Int])]
```
Comment (mikusp) [inline src/Internal/CWrappers.luna]: Docs for another function?

File: native_libs/src/Analysis.cpp [hunk 3]
```diff
@@ -147,12 +147,101 @@ struct Sum : Variance<T>
...
+        else
+            down(value);
+    }
+    void operator() () {}
+
+    double get()
+    {
+        const auto upMean = up.get();
+        const auto downMean = down.get();
+        return 100 * upMean / (upMean - downMean);
```
[the remark this answers is not in the document]
Comment (piotrMocz) [reply native_libs/src/Analysis.cpp]: That's converting a fraction into percentage (like: 0.56 -> 56%).

[the remark this answers is not in the document]
Comment (mwu-tow) [reply native_libs/src/Analysis.cpp]: Basically it is to fit the indicator to the 0-100 range. That's how it is commonly used (see the formulas eg on [wiki](https://en.wikipedia.org/wiki/Relative_strength_index#Calculation).
The formula was described in #36. I hope I understood Python correctly.

**Comment by mwu-tow** [comment]
@mikusp  Thanks for comments, I'll address them tomorrow.

**Pull request merged**